### ISSUE TITLE
C# Installation guide might be out of date

### ISSUE THREAD

**Issue by jdavid82** [wsdjeg/SpaceVim#2935]
Using newly installed macOS Mojave 10.14.5

## Expected behavior, english is required
When I follow the steps shown here: https://spacevim.org/layers/lang/csharp/
Then I should be able to edit and save C# files without seeing error messages.

## The reproduce ways from Vim starting (Required!)
1. When I add "lang#csharp" layer as described here: https://spacevim.org/layers/lang/csharp/
1. And restart vim and watch it automatically clone Omnisharp to $HOME/.cache/vimfiles/repos/github.com/OmniSharp/omnisharp-vim
1. And then when I try to follow the next steps which is navigating to a folder called "server" which should be inside "$HOME/.cache/vimfiles/repos/github.com/OmniSharp/omnisharp-vim".
I am unable to find that "server" folder. Which leads me to believe that the folder structure changed in Ombisharp and the SpaceVim C# guide should be updated.

Also (some extra info just in case), in that OmniSharp repo I found another folder called "installer" but based on the README file it contained and the powershell file it had... it would be a Windows specific file and I'm on a Mac...

Additional information:
1. (Following on from the last step above) I create a new project with "dotnet new console" command.
1. Then I open a generated C# file with vim (Program.cs)
1. Then make small changes to that file and when I try to save it by using the command ```:w``` I was told that OmniSharp wasn't installed and that if I wanted to install it, so I selected ```yes```.
1. Then it proceeded to download and install OmniSharp (I don't know the location where it installed it).
1. Then I closed and re-opened vim
1. Then I made a small change to the C# file again with vim and tried to save it with the command ```:w```
And got the error described in the debug info below

## Debug info

```
"Program.cs" [dos] 20L, 544C written
Error detected while processing /Users/david/.cache/vimfiles/repos/github.com/OmniSharp/omnisharp-vim/autoload/OmniSharp.vim:
line  794:
E122: Function OmniSharp#HighlightEchoKind already exists, add ! to replace it
Error detected while processing BufWritePost Auto commands for "*.cs":
E117: Unknown function: OmniSharp#AddToProject
Press ENTER or type command to continue
```
So in conclusion, I think it all started to go downhill when I was unable to find the "server" folder so that I could run msbuild or xbuild, and instead I let OmniSharp install by itself from within vim which it most likely didn't do it properly... and because I'm a newbie I don't know how to rollback/uninstall that version of Omnisharp...

Any help would be appreciated as I'm keen to try out SpaceVim with C#, thanks.

## Screenshots

No screenshots needed

**Comment by wsdjeg** [comment]
It seems a bug of this layer, I will look into it, thanks for repporting.

________________________________
wsdjeg@outlook.com

**Comment by ChrisTsou** [comment]
Same thing with me. Same error messages when i try to save a file. Didn't find server folder.

**Comment by jdavid82** [comment]
thank you guys
